bootique-shiro-web-jwt/src/main/java/io/bootique/shiro/web/jwt/token/JwtTokenProvider.java

```diff
@@ -0,0 +1,42 @@
+package io.bootique.shiro.web.jwt.token;
+
+import io.bootique.shiro.web.jwt.keys.JwksProvider;
+import io.bootique.shiro.web.jwt.token.claim.JwtClaim;
+import io.bootique.shiro.web.jwt.token.claim.StringListClaim;
+import io.jsonwebtoken.*;
+import io.jsonwebtoken.io.IOException;
+import io.jsonwebtoken.security.Jwk;
+
+import java.security.Key;
+import java.util.List;
+
+public class JwtTokenProvider {
+
+    private final JwksProvider jwkProvider;
+    private JwtClaim<List<String>> rolesClaim;
+
+    public JwtTokenProvider(JwksProvider jwkProvider) {
+        this.jwkProvider = jwkProvider;
+    }
+
+    public void setRolesClaim(StringListClaim claim) {
```

Comment:
This object should be immutable. All initialization should happen in constructor. Also see my earlier comment about casting to `StringListClaim`